web/src/routes/package.js

```diff
@@ -172,7 +283,61 @@ package_router.delete('/:id', async(req,res) => {
 //          - 404: Package does not exist.
 //          - 500: The package rating system choked on at least one of the metrics.
 package_router.get('/:id/rate', async(req,res) => {
+    // This validates that the req.body conforms to the PackageID schema
+    const newPackageIDSchema = new PackageID(req.body)
+    let isValid = true
+    try {
+        await newPackageIDSchema.validate()
+    }
+    catch {
+        isValid = false    
+        // if the format of the input is not in PackageID, return a 400 code
+        res.status(400).json({ message: 'There is missing field(s) in the PackageID or it is formed improperly'})
+    }
+    if(isValid)
+    {
+        // use find by id to find a Metadata schema with PackageID. 404 if it doesn't 
+        const curPackage = await Package.findById({ _id: req.params.id})
+        const curPackageData = await PackageData.findById( curPackage.data )    
```

Comment:
Im not sure if this would cause an error in the case where curPackage does not exist since you are trying to access something that may not exist, might cause page fault/seg fault but if you tested the fail case and it worked then we good

Reply:
If does cause error, should just have to switch order of the 404 check and this line and put this line in the else clause